dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/route/RouteService.java

```diff
@@ -316,33 +333,84 @@ public ResponseEntity<ResponseBodyEmitter> execute(
   }
 
   protected ResponseEntity<Flux<DataBuffer>> retrieve(
-      WebClient.RequestHeadersSpec<?> requestHeadersSpec) {
+      WebClient.RequestHeadersSpec<?> requestHeadersSpec,
+      HttpMethod httpMethod,
+      String upstreamUrl,
+      String routeId,
+      UserDetails userDetails) {
     WebClient.ResponseSpec responseSpec =
         requestHeadersSpec
             .retrieve()
             .onStatus(httpStatusCode -> true, clientResponse -> Mono.empty());
 
+    Audit.AuditBuilder auditBuilder =
+        Audit.builder()
+            .auditScope(AuditScope.API)
+            .createdBy(userDetails.getUsername())
+            .auditType(AuditType.SECURITY);
+
+    RouteRunApiAuditEntity routeRunAuditEntity = new RouteRunApiAuditEntity();
+    routeRunAuditEntity.setSource("Route Run");
```

Comment:
Details... but... can we drop the habit of naming a variable the same as the class - especially if the class name is that long? This can simply be called "entry" or "audit".